kratos/utilities/dof_updater.h

```diff
@@ -0,0 +1,189 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Jordi Cotela
+//
+
+#if !defined(KRATOS_DOF_UPDATER_H_INCLUDED )
+#define  KRATOS_DOF_UPDATER_H_INCLUDED
+
+
+// System includes
+#include <string>
+#include <iostream>
+
+
+// External includes
+
+
+// Project includes
+#include "includes/define.h"
+#include "includes/model_part.h"
+
+namespace Kratos
+{
+///@addtogroup KratosCore
+///@{
+
+///@name Kratos Classes
+///@{
+
+/// Utility class to update the values of degree of freedom (Dof) variables after solving the system.
+/** This class encapsulates the operation of updating nodal degrees of freedom after a system solution.
+ *  In pseudo-code, the operation to be performed is
+ *  for each dof: dof.variable += dx[dof.equation_id]
+ *  This operation is a simple loop in shared memory, but requires additional infrastructure in MPI,
+ *  to obtain out-of-process update data. DofUpdater takes care of both the operation and the eventual
+ *  auxiliary infrastructure.
+ *  @see TrilinosDofUpdater for the trilinos version.
+ */
+template< class TSparseSpace >
+class DofUpdater
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of DofUpdater
+    KRATOS_CLASS_POINTER_DEFINITION(DofUpdater);
+
+    using DofsArrayType = ModelPart::DofsArrayType;
+    using SystemVectorType = typename TSparseSpace::VectorType;
+
+    using UniquePointer = std::unique_ptr<DofUpdater>;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    DofUpdater(){}
+
+    /// Deleted copy constructor
+    DofUpdater(DofUpdater const& rOther) = delete;
+
+    /// Destructor.
+    virtual ~DofUpdater(){}
+
+    /// Deleted assignment operator
+    DofUpdater& operator=(DofUpdater const& rOther) = delete;
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    /// Create a new instance of this class.
+    /** This function is used by the SparseSpace class to create new
+     *  DofUpdater instances of the appropriate type.
+     *  @return a std::unique_pointer to the new instance.
+     *  @see UblasSpace::CreateDofUpdater(), TrilinosSpace::CreateDofUpdater().
+     */
+    virtual UniquePointer Create() const
+    {
+        return UniquePointer(new DofUpdater());
```

Comment:
I think it is better to use `make_unique` here => [see here](https://stackoverflow.com/questions/37514509/advantages-of-using-stdmake-unique-over-new-operator?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa)

Reply:
`make_unique` is C++14, I'm not sure we can support that.

Reply:
actually i think we could define a Kratos::make_unique, which internally calls new, or make_unique is std::make_unique is available